src/simulation_lifecycle/simulation.c

```diff
@@ -32,7 +35,7 @@ int parse_default_sim_config(const cJSON *simulation_config, cJSON *target);
  */
 int write_sim_config(const cJSON *simulation_config, char *config_json);
 
-int build_simulation_scenario(cJSON *simulation_config, Node_t  **data_sources) {
+int build_simulation_scenario(cJSON *simulation_config, feature_set_t *p_features, relation_set_t *p_relations) {
```

Comment:
This function definition is outdated. Please, change it to 
`int build_simulation_scenario(cJSON *simulation_config, node_t  **data_sources) {`